prow/git/v2/client_factory.go

```diff
@@ -285,13 +279,19 @@ func (c *clientFactory) ClientFor(org, repo string) (RepoClient, error) {
 		if err := os.MkdirAll(cacheDir, os.ModePerm); err != nil && !os.IsExist(err) {
 			return nil, err
 		}
+		if err := authenticateCookieFile(c.cookieFilePath, repoClient); err != nil {
+			return nil, err
+		}
 		if err := cacheClientCacher.MirrorClone(); err != nil {
 			return nil, err
 		}
 	} else if err != nil {
 		// something unexpected happened
 		return nil, err
 	} else {
+		if err := authenticateCookieFile(c.cookieFilePath, repoClient); err != nil {
```

Comment:
Why do we need this here? Presumably the first call would have configured it the first time we cloned the repo?

Reply:
Done